simapp/abci.go

```diff
@@ -75,10 +185,63 @@ func (h *VoteExtensionHandler) VerifyVoteExtension() sdk.VerifyVoteExtensionHand
 		case !bytes.Equal(req.Hash, ve.Hash):
 			return &abci.ResponseVerifyVoteExtension{Status: abci.ResponseVerifyVoteExtension_REJECT}, nil
 
-		case len(ve.Data) != 1024:
-			return &abci.ResponseVerifyVoteExtension{Status: abci.ResponseVerifyVoteExtension_REJECT}, nil
 		}
-
 		return &abci.ResponseVerifyVoteExtension{Status: abci.ResponseVerifyVoteExtension_ACCEPT}, nil
 	}
 }
+
+// PreFinalizeBlock -> Extract VEs from previous block and persist in DB.
+func (app *SimApp) NewPreFinalizeBlockHookHandler() sdk.PreFinalizeBlockHook {
+	return func(ctx sdk.Context, req *abci.RequestFinalizeBlock) error {
+		if req.Height > 5 {
+			var ve VoteExtension
+			var eventData *stakingtypes.EventRecord
+			for _, v := range req.Txs {
+				// Have a way to segregate b/w signed VEs and regular txs. Assuming for VEs, byte length would be 8
+				if len(v) >= 8 {
+					err := json.Unmarshal(v, &ve)
+					if err != nil {
+						fmt.Println("Error Occured while unmarshaling in PreFinalize: ", err)
+						continue
+					}
+					if ve.Result == Vote_YES {
+						if err := json.Unmarshal(ve.Data, &eventData); err == nil {
+							fmt.Println("PERSISTING VE DATA!", "HEIGHT: ", ctx.BlockHeight())
+							app.StakingKeeper.SetEventRecord(ctx, *eventData)
+						}
+					}
+				}
+
+			}
+			return nil
+		}
+
+		return nil
+
+	}
+
+}
+
+func (h *VoteExtensionHandler) SetHandlers(bApp *baseapp.BaseApp) {
+	bApp.SetExtendVoteHandler(h.ExtendVote())
+	bApp.SetVerifyVoteExtensionHandler(h.VerifyVoteExtension())
+}
+
+func (app *SimApp) SetProposalHandlers(bApp *baseapp.BaseApp) {
+	bApp.SetPrepareProposal(app.NewPreparePropsalHandler())
+	bApp.SetProcessProposal(app.NewProcessProposalHandler())
+	bApp.SetPreFinalizeBlockHook(app.NewPreFinalizeBlockHookHandler())
+}
+
+func getConfirmedEventLogFromEth() stakingtypes.EventRecord {
```

Comment:
How will this computed in practice? Is it deterministic? Does it change with time? If not, then there is no guarantee that VE/ST will ever be approved (comet continues to generate blocks, though).